docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+
+Before you run any queries, be sure to have ksqlDB start all queries from the earliest point in each topic.
+
+```sql
+SET 'auto.offset.reset' = 'earliest';
+```
+
+### Invoke the scalar function
+
+Inspect the `formula` function by running:
+
+```sql
+DESCRIBE FUNCTION formula;
+```
+
+It should output the following. ksqlDB shows all the type signatures that the UDF implements, which in two in this case.
+
+```
+Name        : FORMULA
+Author      : example user
+Version     : 1.0.2
+Overview    : A custom formula for important business logic.
+Type        : SCALAR
+Jar         : /opt/ksqldb-udfs/example-udfs-0.0.1.jar
+Variations  : 
+
+	Variation   : FORMULA(a DOUBLE, b DOUBLE)
+	Returns     : BIGINT
+	Description : A special variant of the formula, handling double parameters.
+
+	Variation   : FORMULA(a INT, b INT)
+	Returns     : BIGINT
+	Description : The standard version of the formula with integer parameters.
+```
+
+Create a stream named `s1`:
+
+```sql
+CREATE STREAM s1 (
+    a VARCHAR KEY,
+    b INT,
+    c INT
+) WITH (
+    kafka_topic = 's1',
+    partitions = 1,
+    value_format = 'avro'
+);
+```
+
+Insert some rows into the stream:
+
+```sql
+INSERT INTO s1 (a, b, c) VALUES ('k1', 2, 3);
+INSERT INTO s1 (a, b, c) VALUES ('k2', 4, 6);
+INSERT INTO s1 (a, b, c) VALUES ('k3', 6, 9);
+```
+
+Execute a push query. Recall what `formula` does. When given two integers, it multiples the together, then adds the value of the parameter `ksql.functions.formula.base.value`, which is set to `5` in your Docker Compose file:
+
+```sql
+SELECT a, formula(b, c) AS result FROM s1 EMIT CHANGES;
+```
+
+You should see:
+
+```
++--------------------------------------------------------------+--------------------------------------------------------------+
+|A                                                             |RESULT                                                        |
++--------------------------------------------------------------+--------------------------------------------------------------+
+|k1                                                            |11                                                            |
+|k2                                                            |29                                                            |
+|k3                                                            |59                                                            |
+```
+
+Try the other variant which takes two doubles. This implementation takes the ceiling of `a` and `b` before multiplying. Notice how you can use constants instead of column names as arguments to the function:
```

Comment:
```suggestion
Try the other variant of the `formula` function, which takes two doubles. This implementation takes the ceiling of `a` and `b` before multiplying. Notice how you can use constants instead of column names as arguments to the function:
```